packages/examples/cvat/exchange-oracle/pyproject.toml

```diff
@@ -28,19 +28,104 @@ pyinstrument = "^4.6.2"
 
 
 [tool.poetry.group.dev.dependencies]
-black = "^23.1.0"
 pre-commit = "^3.0.4"
-isort = "^5.12.0"
+ruff = "^0.6.0"
 
-[tool.isort]
-profile = "black"
-forced_separate = ["tests"]
-line_length = 100
-skip_gitignore = true # align tool behavior with Black
-
-[tool.black]
+[tool.ruff]
 line-length = 100
-target-version = ['py310']
+target-version = "py310"
+
+
+[tool.ruff.lint]
+select = ["ALL"]
+unfixable = [
+    "RUF005",  # messes up concantenation with numpy structures
+]
+ignore = [
+    "ANN101",  # method args annotations (mypy will take care of that)
+    "ANN001",  # |
+    "ANN202",  # |
+    "ANN201",  # |
+    "ANN401",  # |
+    "ANN102",  # |
+    "RUF001",  # Allow cyrillic letters in comments
+    "COM812",  # Trailing comas are handled by ruff format
+    "ISC001",  # Recommended to be disabled when using ruff format
+    "B904",    # Raise from: modern pythons preserve previous exceptions
+    "EM",      # Forbids using literal strings in exceptions. I think it's ok
+    "TRY003",  # Forbids long messages in exceptions, again, it's ok
+    "G004",    # Logging statement uses f-string — it's ok, f-strings are fast
+    "A003",    # Class attribute `id` is shadowing a Python builtin — it's ok in class body
+    "FIX001",  # Forbids using TODOs, but TODOs are useful
+    "FIX002",  # |
+    "TD001",   # |
+    "TD002",   # |
+    "TD003",   # |
+    "E711",    # Allow == None comparisons for sqlalchemy queries
+    "E712",    # Allow == True comparisons for sqlalchemy queries
+    "PERF203", # Noisy microoptimisation
+    # Want to resolve eventually, but not now:
+    "S101",    # Allow asserts (there are too many of them right now to fix)
+    "TRY401",  # Checks for excesive logging of exception objects
+    "E501",    # Too long lines
```

Comment:
There should be no such lines, as it's enforced by the code formatter. Except some comments, which are an exception in its configuration (e.g. urls).

Reply:
> There should be no such lines, as it's enforced by the code formatter.

Code formatter cannot do anything with comments and strings without changing AST, and it's something that it doesn't do, AFAIK.

> Except some comments, which are an exception in its configuration (e.g. urls).

I'm not exactly sure I follow you here regarding configuration and urls. If you imply there is/should be some way of configuring exceptions in formatter, I believe I've touched upon how this works above.

Since there are quite many exceptions where formatter is unable to step in without potentially changing semantics, I decided to ignore this rule for now. 

<details>
<summary>Existing violations:</summary>

```bash
 $ ruff check . --select E501 --output-format concise
src/crons/process_recording_oracle_webhooks.py:84:101: E501 Line too long (124 > 100)
src/crons/process_recording_oracle_webhooks.py:85:101: E501 Line too long (108 > 100)
src/crons/process_recording_oracle_webhooks.py:91:101: E501 Line too long (135 > 100)
src/crons/process_recording_oracle_webhooks.py:110:101: E501 Line too long (124 > 100)
src/crons/process_recording_oracle_webhooks.py:111:101: E501 Line too long (111 > 100)
src/crons/process_recording_oracle_webhooks.py:130:101: E501 Line too long (135 > 100)
src/crons/state_trackers.py:127:101: E501 Line too long (146 > 100)
src/crons/state_trackers.py:152:101: E501 Line too long (154 > 100)
src/handlers/completed_escrows.py:101:101: E501 Line too long (123 > 100)
src/handlers/job_creation.py:238:101: E501 Line too long (120 > 100)
src/handlers/job_creation.py:506:101: E501 Line too long (124 > 100)
src/handlers/job_creation.py:524:101: E501 Line too long (112 > 100)
src/handlers/job_creation.py:551:101: E501 Line too long (124 > 100)
src/handlers/job_creation.py:559:101: E501 Line too long (145 > 100)
src/handlers/job_creation.py:668:101: E501 Line too long (112 > 100)
src/handlers/job_creation.py:704:101: E501 Line too long (123 > 100)
src/handlers/job_creation.py:711:101: E501 Line too long (115 > 100)
src/handlers/job_creation.py:833:101: E501 Line too long (121 > 100)
src/handlers/job_creation.py:941:101: E501 Line too long (145 > 100)
src/handlers/job_creation.py:963:101: E501 Line too long (126 > 100)
src/handlers/job_creation.py:1020:101: E501 Line too long (111 > 100)
src/handlers/job_creation.py:1222:101: E501 Line too long (103 > 100)
src/handlers/job_creation.py:1564:101: E501 Line too long (112 > 100)
src/handlers/job_creation.py:1607:101: E501 Line too long (132 > 100)
src/handlers/job_creation.py:1693:101: E501 Line too long (111 > 100)
src/handlers/job_creation.py:1718:101: E501 Line too long (126 > 100)
src/handlers/job_creation.py:1726:101: E501 Line too long (139 > 100)
src/handlers/job_creation.py:1888:101: E501 Line too long (133 > 100)
src/handlers/job_creation.py:2008:101: E501 Line too long (152 > 100)
src/handlers/job_creation.py:2031:101: E501 Line too long (127 > 100)
src/handlers/job_creation.py:2409:101: E501 Line too long (105 > 100)
src/handlers/job_creation.py:2410:101: E501 Line too long (107 > 100)
tests/integration/chain/test_kvstore.py:162:101: E501 Line too long (114 > 100)
tests/integration/chain/test_kvstore.py:172:101: E501 Line too long (109 > 100)
tests/integration/cron/test_process_job_launcher_webhooks.py:303:101: E501 Line too long (105 > 100)
tests/integration/cron/test_process_recording_oracle_webhooks.py:196:101: E501 Line too long (103 > 100)
tests/integration/services/test_webhook.py:86:101: E501 Line too long (104 > 100)
tests/integration/services/test_webhook.py:161:101: E501 Line too long (104 > 100)
tests/utils/constants.py:26:101: E501 Line too long (146 > 100)
tests/utils/constants.py:35:101: E501 Line too long (159 > 100)
Found 40 errors.
```
</details>

Reply:
Ok, but there are cases where the lines (strings, actually) became long because of changes in this PR.

Reply:
Good point 😄. I've fixed existing violations and re-enabled the rule.